### PR TITLE
Feat: Tutorials for DashBoard, MapDesigner and History Pages

### LINKED ISSUE #18: Tutorials

### PR DESCRIPTION
**Pull request by bhoopesh369**

### CODE REVIEW

**Comment by bhoopesh369** [comment]
Todo: Integrate With Backend

File: src/components/BattleTV/BattleTV.tsx [hunk 1]
```diff
@@ -200,13 +206,20 @@ function PaginatedItems() {
 }
 export default function BattleTV(): JSX.Element {
   const { setIsOpen } = useTour();
+  const currentUserapi = new CurrentUserApi(apiConfig);
 
   useEffect(() => {
-    setIsOpen(true);
+    setTimeout(() => {
+      currentUserapi.getCurrentUser().then(response => {
```
Comment (vigneshd332) [inline src/components/BattleTV/BattleTV.tsx:213]: use `&&` here in a single if statement

File: src/components/CommitHistory/HistoryMain/History.tsx [hunk 2]
```diff
@@ -43,13 +44,19 @@ export default function History(): JSX.Element {
...
+  const currentUserapi = new CurrentUserApi(apiConfig);
 
   const { setIsOpen } = useTour();
 
   useEffect(() => {
     setTimeout(() => {
-      setIsOpen(true);
-    }, 500);
+      currentUserapi.getCurrentUser().then(response => {
+        if (response.isTutorialComplete === false) {
```
Comment (vigneshd332) [inline src/components/CommitHistory/HistoryMain/History.tsx]: change this to a single if statement everywhere

File: src/pages/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -284,16 +285,25 @@ export default function Dashboard(): JSX.Element {
     }
   };
 
-  const currentUserapi = new CurrentUserApi();
+  const currentUserapi = new CurrentUserApi(apiConfig);
+
+  const User = useAppSelector(user);
+  const Navigate = useNavigate();
```
Comment (vigneshd332) [inline src/pages/Dashboard/Dashboard.tsx]: use camelcase please

**Review by vigneshd332 — CHANGES_REQUESTED** [review]

File: src/pages/Dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -309,281 +310,287 @@ export default function Dashboard(): JSX.Element {
...
-                        className={styles.hoverIcon}
-                      />
+                    <div className={styles.challName}>
+                      {dailyChallenge.challName}
                     </div>
-                    <div className={styles.settingsIcon} id="Shortcuts">
+                    <div className={styles.infoIcon}>
                       <FontAwesomeIcon
-                        title={'Shortcuts'}
+                        title={'Shorcuts'}
```
Comment (vigneshd332) [inline src/pages/Dashboard/Dashboard.tsx]: Why undo this lol

File: src/components/TourIntroModal/TourIntroModal.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        console.log('Tutorial level updated');
+      })
+      .catch(err => {
+        if (err instanceof ApiError) Toast.error(err.message);
+      });
+  };
+
+  useEffect(() => {
+    currentUserApi.getCurrentUser().then(res => {
+      if (res.isTutorialComplete === false && res.tutorialLevel === 1) {
```
Comment (vigneshd332) [inline src/components/TourIntroModal/TourIntroModal.tsx]: === false could just be one !

File: src/components/TourIntroModal/TourIntroModal.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+import { useNavigate } from 'react-router-dom';
+import { useTour } from '@reactour/tour';
+
+const TourIntroModal = (): JSX.Element => {
+  const [isTourOpen, setIsTourOpen] = useState(false);
+  const currentUserApi = new CurrentUserApi(apiConfig);
+  const User = useAppSelector(user);
+  const navigate = useNavigate();
+
+  const newLocal = 'fw-bold fs-3';
```
Comment (vigneshd332) [inline src/components/TourIntroModal/TourIntroModal.tsx]: what is the use of this variable? you can directly specify as string no

File: src/components/TourIntroModal/TourIntroModal.module.css [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+.showBtn {
+    display: flex;
+    justify-content: end;
+    align-items: center;
+}
+
+.btn{
+    width: 120px;
+    height: 42px;
+}
```
Comment (vigneshd332) [inline src/components/TourIntroModal/TourIntroModal.module.css]: EOF bro

File: src/pages/Dashboard/Dashboard.tsx [hunk 8]
```diff
@@ -309,281 +310,287 @@ export default function Dashboard(): JSX.Element {
...
+                    <Button
+                      className={styles.closeEditorButton}
+                      onClick={() => {
+                        updateDividerPosition(dividerPosition - 1);
+                      }}
+                      variant="dark"
+                    >
+                      <FontAwesomeIcon
+                        size={'sm'}
+                        icon={faChevronLeft as IconProp}
```
Comment (vigneshd332) [inline src/pages/Dashboard/Dashboard.tsx:463]: you can remove the common buttons outside the if statement here, will save on code

**Review by vigneshd332 — CHANGES_REQUESTED** [review]

File: src/pages/Dashboard/Dashboard.tsx (hunk 4 above)
Comment (bhoopesh369) [reply src/pages/Dashboard/Dashboard.tsx:501]: bruh its shortcuts only in the pr.

File: src/pages/Dashboard/Dashboard.tsx (hunk 8 above)
Comment (bhoopesh369) [reply src/pages/Dashboard/Dashboard.tsx:463]: here It cant be made common 
because of react bootstarp buutton toolbar.
Thats why Shubam made it seperate.

**Comment by vigneshd332** [comment]
Use an ENUM type for dailychallegne/dashboard page type

File: packages/map-designer/src/MapDesignerComponent.tsx [hunk 9]
```diff
@@ -117,7 +121,10 @@ export default function MapDesignerComponent(
             </OverlayTrigger>
           ))}
         </ButtonGroup>
-        <ButtonGroup style={{ pointerEvents: 'auto', padding: '1rem 1rem 0' }}>
+        <ButtonGroup
+          style={{ pointerEvents: 'auto', padding: '1rem 1rem 0' }}
```
Comment (vigneshd332) [inline packages/map-designer/src/MapDesignerComponent.tsx:125]: inline css? 🤨

File: src/components/MapDesigner/MapDesigner.module.css [hunk 10]
```diff
@@ -63,3 +63,14 @@
...
+
+.PsuedoMap{
+  position: absolute;
+  top:60%;
+  left:50%;
+  transform: translate(-50%,-50%);
+  width: 50vw;
+  height: 70vh;
+  background: transparent;
+  pointer-events: none;
```
Comment (vigneshd332) [inline src/components/MapDesigner/MapDesigner.module.css:75]: EOF

**Review by vigneshd332 — CHANGES_REQUESTED** [review]

**Review by vigneshd332 — APPROVED** [review]

**Review by Ram-20062003 — APPROVED** [review]

**Pull request merged**